tests/views/schema.py

```diff
@@ -20,10 +21,121 @@ def has_permission(self, source: Any, info: Info[Any, Any], **kwargs: Any) -> bo
         return False
 
 
+class ConditionalFailPermission(BasePermission):
+    @property
+    def message(self):
+        return f"failed after sleep {self.sleep}"
+
+    async def has_permission(self, source, info, **kwargs: Any) -> bool:
+        self.sleep = kwargs.get("sleep", None)
+        self.fail = kwargs.get("fail", True)
+        if self.sleep is not None:
+            await asyncio.sleep(kwargs["sleep"])
+        return not self.fail
+
+
 class MyExtension(SchemaExtension):
+    # a counter to keep track of how many operations are active
+    active_counter = 0
+
     def get_results(self) -> Dict[str, str]:
         return {"example": "example"}
 
+    def resolve(self, _next, root, info: Info, *args: Any, **kwargs: Any):
+        self.active_counter += 1
+        try:
+            self.resolve_called()
+            return _next(root, info, *args, **kwargs)
+        finally:
+            self.active_counter -= 1
+
+    def resolve_called(self):
+        pass
+
+    def lifecycle_called(self, event, phase):
+        pass
+
+    def on_operation(self):
+        self.lifecycle_called("operation", "before")
+        self.active_counter += 1
+        yield
+        self.lifecycle_called("operation", "after")
+        self.active_counter -= 1
+
+    def on_validate(self):
+        self.lifecycle_called("validate", "before")
+        self.active_counter += 1
+        yield
+        self.lifecycle_called("validate", "after")
+        self.active_counter -= 1
+
+    def on_parse(self):
+        self.lifecycle_called("parse", "before")
+        self.active_counter += 1
+        yield
+        self.lifecycle_called("parse", "after")
+        self.active_counter -= 1
+
+    def on_execute(self):
+        self.lifecycle_called("execute", "before")
+        self.active_counter += 1
+        yield
+        self.lifecycle_called("execute", "after")
+        self.active_counter -= 1
+
+
+class MyAsyncExtension(SchemaExtension):
```

Comment:
Can this use https://github.com/strawberry-graphql/strawberry/blob/b7982878f1b4ec73d74af4a86ab85fdeb921763b/tests/schema/extensions/test_extensions.py#LL194C7-L194C23 ?